### PR TITLE
Optimize calls on group details page, update group settings based on access type, delete group

### PR DESCRIPTION
**Pull request by juliasliu**
Haven't refactored code yet, will do that after I finish other MVP 3 features

### CODE REVIEW

**Comment by juliasliu** [comment]
UI is not finalized: not sure whether having the removing and adding member button on the hover card or directly in the circle visualization is more intuitive for users to understand.

File: src/main/webapp/scripts/group-constants.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
+// Default public settings object
+var publicSettings = {
```
Comment (phillipdold) [inline src/main/webapp/scripts/group-constants.js:2]: Nice on splitting out the constants.

**Review by phillipdold — COMMENTED** [review]
I am having trouble when using the User level filter and the group filter. It seems to spin for a while. Did you have any trouble when using the filter?

**Comment by juliasliu** [comment]
Hi Phillip! It works on my end, though it does spin for a longer time than when it first loads the page, probably due to the API call and the fact that it is not returning all the groups in the domain so the DFS is making more inner API calls as well. Does it not stop spinning at all?

**Comment by phillipdold** [comment]
> Hi Phillip! It works on my end, though it does spin for a longer time than when it first loads the page, probably due to the API call and the fact that it is not returning all the groups in the domain so the DFS is making more inner API calls as well. Does it not stop spinning at all?

Actually, it just seems to be when I select my user. The other load fine.

**Comment by juliasliu** [comment]
Is this what you see?
Screenshot 2020-07-31 at 2 06 19 PM

**Review by phillipdold — APPROVED** [review]

**Pull request merged**